src/database/DataTypes/Location.js

```diff
@@ -6,12 +6,71 @@
 import Realm from 'realm';
 
 export class Location extends Realm.Object {
-  // TODO
-  // getBreaches(database) <- ('SensorLogs') filter this ID and isInBreach = true
-  // getBatches ?
+  // TODO remove data object
+  data = {
+    isInBreach: false,
+    totalStock: 16,
+    currentTemperature: '5.2',
+    temperatureExposure: { minTemperature: 2.5, maxTemperature: 10 },
+    temperaturePoints: {
+      maxLine: [
+        { date: 'Feb 23', temp: 5 },
+        { date: 'Feb 24', temp: 8 },
+        { date: 'Feb 25', temp: 8.5 },
+        { date: 'Feb 26', temp: 9 },
+        { date: 'Feb 27', temp: 10 },
+        { date: 'March 1', temp: 7 },
+        { date: 'March 2', temp: 5 },
+        { date: 'March 3', temp: 4 },
+        { date: 'March 4', temp: 5 },
+        { date: 'March 5', temp: 6 },
+      ],
+      minLine: [
+        { date: 'Feb 23', temp: 2.5 },
+        { date: 'Feb 24', temp: 5 },
+        { date: 'Feb 25', temp: 5.7 },
+        { date: 'Feb 26', temp: 7.5 },
+        { date: 'Feb 27', temp: 8 },
+        { date: 'March 1', temp: 6 },
+        { date: 'March 2', temp: 3.4 },
+        { date: 'March 3', temp: 3 },
+        { date: 'March 4', temp: 2.5 },
+        { date: 'March 5', temp: 4 },
+      ],
+      hazards: [{ date: 'Feb 27', temp: 10, onClick: () => console.log('CLICKED HAZARD 4') }],
+      minTemp: 2,
+      maxTemp: 8,
+    },
+  };
 
   get isFridge() {
-    return this.locationType && this.locationType.description.toLowerCase() === 'fridge';
+    if (!this.locationType) return null;
+    if (!this.locationType.description) return null;
+    return String(this.locationType.description).toLowerCase() === 'fridge';
+  }
+
+  getTemperaturePoints() {
+    return this.data.temperaturePoints;
```

Comment:
i've been working on this branch, with helper methods: https://github.com/openmsupply/mobile/compare/%23874-Vaccine-Module-Page...%23874-Refinements%5D-Vaccine-Modul-Page
untested by the way, what do you think though ?

Reply:
@andreievg 

Is there a case for opening a breach modal for 'all breaches' for a location? Rather than *only* clicking on a hazard icon? If not, could make one [View Breaches] button, or make the number of breaches clickable to open the modal. 

Then, return all the breaches from this method rather than just the length and store them on mounting. Shouldn't be more logic to be able to do this, same as opening a breach modal from clicking a hazard icon, just more breaches.